tools/cudaq-qpud/RestServerMain.cpp

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#include "common/Registry.h"
+#include "common/RemoteKernelExecutor.h"
+#include "llvm/Support/CommandLine.h"
+
+// Declare CUDAQ MPI API that we need since we cannot compile with cudaq.h
+// without RTTI (needed to link this tool against LLVMSupport).
+namespace cudaq {
+namespace mpi {
+void initialize();
+void finalize();
+} // namespace mpi
+} // namespace cudaq
+
+//===----------------------------------------------------------------------===//
+// Command line options.
+//===----------------------------------------------------------------------===//
+
+static llvm::cl::opt<int>
+    port("port", llvm::cl::desc("TCP/IP port that the server will listen to."),
+         llvm::cl::init(3030));
+
+int main(int argc, char **argv) {
+  llvm::cl::ParseCommandLineOptions(argc, argv, "CUDA Quantum REST server\n");
+  cudaq::mpi::initialize();
+  auto restServer = cudaq::registry::get<cudaq::RemoteRuntimeServer>("rest");
```

Comment:
If the tool uses the Servers, should all server related interface and implementations be here in this folder?

Reply:
The client and server impls both depend on the Json schema (`JsonConvert.h`) so were placed in the same spot.
Should we move the Json schema to a common location and move the client and server pieces to their use (the QPU implementation and `cudaq-qpud` tool?

Reply:
Discussed offline with @amccaskey and we agreed that this refactoring should be deferred as a follow-up.